ethstorage/storage_manager.go

```diff
@@ -108,8 +114,15 @@ func (s *StorageManager) DownloadFinished(newL1 int64, kvIndices []uint64, blobs
 		}
 	}
 
+	lastKvIdx, err := s.l1Source.GetStorageLastBlobIdx(newL1)
+	if err != nil {
+		return err
+	}
+	s.lastKvIdx = lastKvIdx
 	s.localL1 = newL1
 
+	s.updateLocalMetas(kvIndices, commits)
```

Comment:
We should remove the metas in the map if lastKvIdx is smaller because of removal (or at least a TODO here).